src/Builders/ComponentParameterBuilder.cs

```diff
@@ -0,0 +1,351 @@
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add<TValue>(Expression<Func<TComponent, RenderFragment<TValue>?>> parameterSelector, RenderFragment<TValue> template)
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (template is null)
+            {
+                throw new ArgumentNullException(nameof(template));
+            }
+
+            var (name, isCascading) = GetDetailsFromExpression(parameterSelector);
+            return AddParameterToList(name, template, isCascading);
+        }
+
+        /// <summary>
+        /// Add a strongly typed <see cref="RenderFragment{TValue}" /> parameter with a markupFactory for a component under test.
+        /// </summary>
+        /// <typeparam name="TValue">The generic value type</typeparam>
+        /// <param name="parameterSelector">The parameter selector which defines the parameter to add</param>
+        /// <param name="markupFactory">A markup factory that takes a <typeparamref name="TValue"/> as input and returns markup/HTML.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add<TValue>(Expression<Func<TComponent, RenderFragment<TValue>?>> parameterSelector, Func<TValue, string> markupFactory)
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (markupFactory is null)
+            {
+                throw new ArgumentNullException(nameof(markupFactory));
+            }
+
+            return Add(parameterSelector, value => renderTreeBuilder => renderTreeBuilder.AddMarkupContent(0, markupFactory(value)));
+        }
+
+        /// <summary>
+        /// Add a strongly typed <see cref="EventCallback" /> parameter with a callback for a component under test.
+        /// </summary>
+        /// <param name="parameterSelector">The parameter selector which defines the parameter to add</param>
+        /// <param name="callback">A callback that returns a <see cref="Task"/>.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EC>> parameterSelector, Func<Task> callback)
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (callback is null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            return Add(parameterSelector, EC.Factory.Create(this, callback));
+        }

```

Comment:
```suggestion
        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EC>> parameterSelector, Func<Task> callback) => Add(parameterSelector, EC.Factory.Create(this, callback));
```
I think this should be OK.